### ISSUE TITLE
Mongoose Schema fields options

### ISSUE THREAD

**Issue by zabojad** [clemos/haxe-js-kit#102]
Hi !

Is this kind of Schema declarations supported ?
```
var mongoose = require('mongoose');

var UserSchema = new mongoose.Schema({
  username: {
    type: String,
    unique: true,
    required: true
  },
  password: {
    type: String,
    required: true
  }
});

// Export the Mongoose model
module.exports = mongoose.model('User', UserSchema);
```

If yes, how can we do this ?

If not how could we add this ? :)

Thanks !

**Comment by clemos** [comment]
Obviously, you can't "directly" you `module.exports` and such in Haxe.
But you can totally do:
```haxe
var UserSchema = new js.npm.mongoose.Schema({
  username: {
    type: "String", // strings are usually safer here
    unique: true,
    required: true
  },
  password: {
    type: "String",
    required: true
  }
});
var userModel = js.npm.Mongoose.mongoose.model('User', userSchema);
```
This is the "untyped", extern-only way of doing it: you won't have type safety when using this model (documents won't be typed automatically as `{ username : String, password : String }`).

There is an alternative way of declaring schemas, using `js.npm.mongoose.macro` package.
It is shown in this example : https://github.com/clemos/haxe-js-kit/blob/develop/test/MongooseTest.hx
Basically, you use Haxe `typedef`s to define your schema; the typedef (ie: `StuffData`) is transformed into a mongoose schema definition by a macro, and the fields are "copied" to the Model definition (ie: `StuffModel`) to provide type safety.
Regarding `unique` and `required` flags, you can add them as metadata in your typedef: 
```haxe
typedef UserData = {
   @:unique @:required var username : String;
   @:unique @:required var password : String;
};
```
see https://github.com/clemos/haxe-js-kit/blob/develop/util/Mongoose.hx#L435-L485

Let me know if the example is clear enough for you to adapt it to your needs, and if it's what you would expect.

**Comment by zabojad** [comment]
I missed those metadata. Thank you very much, your answer is very clear, it is what I was expecting...
